### ISSUE TITLE
New version of the compiler error, the need to modify system files?

### ISSUE THREAD

**Issue by lenct** [cheat-engine/cheat-engine#951]
![TIM截图20191025182821](https://user-images.githubusercontent.com/44138520/67564536-bdcb2a00-f755-11e9-897b-527fcffd2ac2.png)
![TIM截图20191025182849](https://user-images.githubusercontent.com/44138520/67564540-be63c080-f755-11e9-9e62-eebb4e1e163d.png)

Lazarus, 2.01 to 2.04, compilation is failure

**Comment by lenct** [comment]
And "MIN" function also can produce false match

**Comment by mgrinzPlayer** [comment]
duplicate of:
https://github.com/cheat-engine/cheat-engine/issues/822
https://github.com/cheat-engine/cheat-engine/issues/885
https://github.com/cheat-engine/cheat-engine/issues/918
https://github.com/cheat-engine/cheat-engine/issues/926

**Comment by lenct** [comment]
> duplicate of:
> #822
> #885
> #918
> #926

https://github.com/cheat-engine/cheat-engine/issues/951#issuecomment-546327792

**Comment by lenct** [comment]
Thank you very much! I'm sorry I too long without coming to the BBS

**Comment by frustreated** [comment]
https://github.com/mgrinzPlayer/CheatEngineUpDeluxe
use this to compile ,but i tried ,there still have or may be have errors.
用这个编译，不过我尝试了下，仍旧有报错
from my computer error as follows
我的如下报错
```

error, find this
Building project "CheatEngine 32bit"
系统找不到指定的路径。
```

```
Building project "CheatEngine 64bit"
系统找不到指定的路径。
请按任意键继续. . .
```
other files are compiled OK
其他的都正常编译

**Comment by mgrinzPlayer** [comment]
@frustreated you have to launch Lazarus IDE at least once (open lazarus folder, launch lazarus.exe, click "Upgrade" button, then launch IDE) then you can double click "prepare CE.cmd"

**Comment by frustreated** [comment]
@mgrinzPlayer 
3Q
the only thing i forget to do is click upgrade!!!
@mgrinzPlayer 
3Q very much!

**Comment by frustreated** [comment]
重新尝试编译了下，用的最新版的lazarus，稍微修改点代码即可。
https://github.com/frustreated/cheatengine7.1

**Comment by frustreated** [comment]
> And "MIN" function also can produce false match

yes ,and i tried to modify it to  if else to get the min number and eventully ,compiled success

**Comment by mgrinzPlayer** [comment]
you do not need if else. Just change "lua_objlen(L, 1)" to "integer(lua_objlen(L, 1))"

**Comment by frustreated** [comment]
> you do not need if-then-else. Just change "lua_objlen(L, 1)" to "integer(lua_objlen(L, 1))"

thanks so much , i realy do not know any of larus..
